src/ExactDiagonalization/basis_set_representation.jl

```diff
@@ -1,169 +1,16 @@
-        for (j, v) in col
-            iszero(v) && continue
-            push!(is, i)
-            push!(js, j)
-            push!(vs, v)
-        end
-    end
-
-    matrix = sparse(js, is, vs, length(adds), length(adds))
-    if sort
-        perm = sortperm(adds; kwargs...)
-        return permute!(matrix, perm, perm), permute!(adds, perm)
-    else
-        return matrix, adds
-    end
-end
-
-"""
-    build_basis(
-        ham, address=starting_address(ham);
-        cutoff, filter, sizelim, sort=false, kwargs...
-    ) -> basis
-    build_basis(ham, addresses::AbstractVector; kwargs...)
-
-Get all basis element of a linear operator `ham` that are reachable (via
-non-zero matrix elements) from the address `address`, returned as a vector.
-Instead of a single address, a vector of `addresses` can be passed.
-Does not return the matrix, for that purpose use [`BasisSetRepresentation`](@ref).
-
-Providing an energy cutoff will skip addresses with diagonal elements greater
-than `cutoff`. Alternatively, an arbitrary `filter` function can be used instead.
-Addresses passed as arguments are not filtered.
-A maximum basis size `sizelim` can be set which will throw an error if the expected
-dimension of `ham` is larger than `sizelim`. This may be useful when memory may be a
-concern. These options are disabled by default.
-
-Setting `sort` to `true` will sort the basis. Any additional keyword arguments
-are passed on to `Base.sort!`.
-"""
-function build_basis(
-    ham, addr_or_vec=starting_address(ham);
-    cutoff=nothing,
-    filter=isnothing(cutoff) ? nothing : (a -> diagonal_element(ham, a) ≤ cutoff),
-    sort=false,
-    max_size=Inf, # retained for backwards compatibility; use sizelim instead
-    sizelim=max_size,
-    kwargs...
-)
-    check_address_type(ham, addr_or_vec)
-    single_addr = addr_or_vec isa Union{AbstractArray,Tuple} ? addr_or_vec[1] : addr_or_vec
-    if dimension(ham, single_addr) > sizelim
-        throw(ArgumentError("dimension larger than sizelim"))
-    end
-    # Set up `adds` as queue of addresses. Also returned as the basis.
-    adds = addr_or_vec isa Union{AbstractArray,Tuple} ? [addr_or_vec...] : [addr_or_vec]
-    known_basis = Set(adds)     # known addresses
-
-    i = 0
-    while i < length(adds)
-        i += 1
-        add = adds[i]
-
-        for (off, v) in offdiagonals(ham, add)
-            (iszero(v) || off in known_basis) && continue   # check if valid
-            push!(known_basis, off)
-            !isnothing(filter) && !filter(off) && continue  # check filter
-            push!(adds, off)
-        end
-    end
-
-    if sort
-        return sort!(adds, kwargs...)
-    else
-        return adds
-    end
-end
-
 """
     BasisSetRepresentation(
         hamiltonian::AbstractHamiltonian, addr=starting_address(hamiltonian);
-        sizelim=10^6, nnzs, cutoff, filter, sort=false, kwargs...
+        sizelim=10^8, cutoff, filter, max_depth, stop_after, sort=false, kwargs...
```

Comment:
`nnzs` is not listed here but is documented below, while `max_depth` and `stop_after` appear here but are not explained below

Reply:
Yes, the list of keyword arguments that are accepted and their defaults needs to be checked and updated.

Also, can we please set `sizelim=10^7`?